src/main/main-window.ts

```diff
@@ -1,97 +1,115 @@
-            console.error(details);
-        });
-
-        this.window.webContents.setWindowOpenHandler(({ url }) => {
-            shell.openExternal(url);
-            return { action: "deny" };
-        });
-
-        this.window.webContents.session.webRequest.onBeforeSendHeaders((details, callback) => {
-            callback({ requestHeaders: { Origin: "*", ...details.requestHeaders } });
-        });
-
-        this.window.webContents.session.webRequest.onHeadersReceived((details, callback) => {
-            const obj = { responseHeaders: { ...details.responseHeaders } };
-            if (!obj.responseHeaders["Access-Control-Allow-Origin"] && !obj.responseHeaders["access-control-allow-origin"]) {
-                obj.responseHeaders["Access-Control-Allow-Origin"] = ["*"];
-            }
-            callback(obj);
-        });
-
-        watch(
-            () => this.settings.model.displayIndex,
-            (displayIndex) => this.setDisplay(displayIndex)
-        );
-
-        watch(
-            () => this.settings.model.fullscreen,
-            (fullscreen) => {
-                this.window.setFullScreen(fullscreen);
-                this.window.maximize();
-            }
-        );
-
-        if (process.env.ELECTRON_RENDERER_URL) {
-            this.window.loadURL(process.env.ELECTRON_RENDERER_URL);
-            this.window.webContents.openDevTools();
-        } else {
-            this.window.loadFile(path.join(__dirname, "../renderer/index.html"));
-        }
+import { settingsService } from "./services/settings.service";
+import { logger } from "./utils/logger";
+import { replayContentAPI } from "@main/content/replays/replay-content";
+
+declare const MAIN_WINDOW_VITE_DEV_SERVER_URL: string;
+declare const MAIN_WINDOW_VITE_NAME: string;
+
+const log = logger("main-window");
+
+export function createWindow() {
+    const settings = settingsService.getSettings();
+    log.info("Creating main window with settings: ", settings);
+
+    const primaryDisplay = screen.getPrimaryDisplay();
+    const mainWindow = new BrowserWindow({
+        title: "Beyond All Reason",
+        fullscreen: settings.fullscreen,
+        frame: true,
+        show: true,
+        minWidth: 1440,
+        minHeight: 900,
+        paintWhenInitiallyHidden: true,
+        webPreferences: {
+            preload: path.join(__dirname, "../build/preload.js"),
+            zoomFactor: zoomFactor(primaryDisplay),
+        },
+    });
+    process.env.MAIN_WINDOW_ID = mainWindow.id.toString();
+
+    log.debug("Settings: ", settings);
+
+    mainWindow.once("ready-to-show", () => {
+        mainWindow.setMenuBarVisibility(false);
+        mainWindow.show();
+        mainWindow.focus();
+    });
+
+    mainWindow.webContents.on("render-process-gone", (event, details) => {
+        console.error(details);
+    });
+
+    mainWindow.webContents.setWindowOpenHandler(({ url }) => {
+        shell.openExternal(url);
```

Comment:
I wouldn't do that https://benjamin-altpeter.de/shell-openexternal-dangers/

Reply:
What was I thinking...

Would be grateful if you double check this change :
https://github.com/beyond-all-reason/bar-lobby/pull/277/commits/e6d2a0ffd1515a378746c77af2d5d3176f89997a

Reply:
Hmm, I believe you still need to have `mainWindow.webContents.setWindowOpenHandler` with `{ action: 'deny' }` as without that it will open new window, that we likely ~never want.

I do it in similar way in launcher where I know it's 100% single page application: https://github.com/beyond-all-reason/spring-launcher/blob/2c0a6e17b50ab04ea592c53963c571e5ccb071ba/src/update_dialog.js#L49-L68

Reply:
I had a bad understanding of this. I simply deny everything now.